qiskit_machine_learning/connectors/torch_connector.py

```diff
@@ -192,6 +192,10 @@ def __init__(self, neural_network: NeuralNetwork,
         """
         super().__init__()
         self._neural_network = neural_network
+
+        # turn on input gradients if they are off by default
+        self._neural_network.input_gradients = True
```

Comment:
Do we expect users to possible re-use neural networks. As soon as you use it with the torch connector here it is going to have its flag forced True - so it it started out False the network will no longer behave as it did before if re-used elsewhere - if I understand things

Reply:
Yes, a network maybe and probably should be re-used. The problem is that it is hard to trace when a network is not used any more and gradients may be turned off again. Management of gradients is up to a upper level class in the hierarchy, e.g. a classifier, regressor, or torch connector. But even in this case it is not obvious. I was thinking of this problem when I was writing the code. Unfortunately I don't have a good solution here.